### PR TITLE
enclave-agent: move to clap v3 Derive API

### PR DESCRIPTION
**Pull request by mythi**
clap v2.33.3 is three years old and automated dependabot updates won't work anymore.

Rework enclave-agent command line args handling to the latest Derive API from clap and move to v4.4.6.

### CODE REVIEW

File: src/enclave-agent/src/main.rs [hunk 1]
```diff
@@ -1,90 +1,60 @@
...
-    };
+    let cli = Cli::parse();
 
-    let dc = if let Some(file_path) = matches.value_of("decrypt-config") {
-        DecryptConfig::load_from_file(&file_path.to_string())?
-    } else {
-        DecryptConfig::default()
+    let dc = match DecryptConfig::load_from_file(&cli.decrypt_config) {
+        Ok(config) => config,
+        _ => DecryptConfig::default(),
```
Comment (Xynnn007) [inline src/enclave-agent/src/main.rs:40]: Do we need to raise a warning or info here to tell the caller that the given `DecryptConfig` cannot be loaded and downgrade to a default one?

Comment (mythi) [reply src/enclave-agent/src/main.rs:40]: makes sense, will change

Comment (mythi) [reply src/enclave-agent/src/main.rs]: updated.

File: src/enclave-agent/src/config.rs [hunk 2]
```diff
@@ -33,7 +31,7 @@ pub struct OcicryptConfig {
 impl OcicryptConfig {
     pub fn new(path: String) -> Result<Self, anyhow::Error> {
         if !Path::new(&path).exists() {
-            return Err(anyhow!("ocicrypt config not found"));
+            return Err(anyhow!("ocicrypt config {} does not exist.", path));
```
Comment (Xynnn007) [inline src/enclave-agent/src/config.rs:34]:
```suggestion
            bail!("ocicrypt config {path} does not exist.");
```

File: src/enclave-agent/src/main.rs [hunk 3]
```diff
@@ -1,90 +1,63 @@
...
+    let cli = Cli::parse();
 
-    let dc = if let Some(file_path) = matches.value_of("decrypt-config") {
-        DecryptConfig::load_from_file(&file_path.to_string())?
-    } else {
-        DecryptConfig::default()
+    let dc = match DecryptConfig::load_from_file(&cli.decrypt_config) {
+        Ok(config) => config,
+        Err(e) => {
+            println!("Setting default DecryptConfig because: {:?}", e);
```
Comment (Xynnn007) [inline src/enclave-agent/src/main.rs:41]: Shall we use `warn!("...")` here? As we already import dep `log` now.

**Comment by mythi** [comment]
@Xynnn007 looks OK now?

**Review by Xynnn007 — APPROVED** [review]

**Pull request merged**